packages/frontend/core/src/blocksuite/ai/_common/chat-actions-handle.ts

```diff
@@ -201,94 +204,148 @@
   });
 }
 
-const REPLACE_SELECTION = {
-  icon: ReplaceIcon({ width: '20px', height: '20px' }),
-  title: 'Replace selection',
+/**
+ * Get selected blocks based on current selections
+ * @param host Editor host
+ * @param currentSelections Current selections
+ * @returns Selected blocks and selection state
+ */
+async function getSelectedBlocksForInsert(
+  host: EditorHost,
+  currentSelections: Selections
+): Promise<{ selectedBlocks: BlockComponent[]; isNothingSelected: boolean }> {
+  const currentTextSelection = currentSelections.text;
+  const currentBlockSelections = currentSelections.blocks;
+  const currentImageSelections = currentSelections.images;
+
+  if (!host.doc.root) {
+    return { selectedBlocks: [], isNothingSelected: true };
+  }
+
+  const [_, { isNothingSelected }] = host.command.exec(
+    isNothingSelectedCommand,
+    {
+      currentTextSelection,
+      currentBlockSelections,
+      currentImageSelections,
```

Comment:
No need to pass selection params?

Reply:
This is the previous logic; keep it for now.